domains/Domain.ml

```diff
@@ -41,4 +41,11 @@ sig
   val compress : t -> t
   val print : Format.formatter -> t -> unit
 
+  val print_graphviz_dot : Format.formatter -> t -> unit
+
+  val left_narrow: t -> t -> t
+  val reset_until: t -> t -> t -> t
```

Comment:
I changed the name to 'until' for the moment. Currently I don't have a better name either. 

The 'until' functoin can be used to replace 'reset' with the exception of the 'mask' parameter that's currently supported by 'reset'. I'm not sure how to incorporate that feature into my 'until' function and if it would even make sense.